src/lib/inbox/inbox.ts

```diff
@@ -276,4 +314,101 @@ export class InboxTools {
       overrides || {}
     )
   }
+
+  /**
+   * Send l2 signed tx using delayed inox, which won't alias the sender's adddress
+   * It will be automatically included by the sequencer on l2, if it isn't included
+   * within 24 hours, you can force include it
+   * @param signedTx A signed transaction which can be sent directly to network,
+   *                you can call inboxTools.signL2Message to get.
+   * @returns The l1 delayed inbox's transaction itself.
+   */
+  public async sendL2SignedTx(
+    signedTx: string
+  ): Promise<ContractTransaction | null> {
+    const delayedInbox = IInbox__factory.connect(
+      this.l2Network.ethBridge.inbox,
+      this.l1Signer
+    )
+
+    const sendData = ethers.utils.solidityPack(
+      ['uint8', 'bytes'],
+      [ethers.utils.hexlify(InboxMessageKind.L2MessageType_signedTx), signedTx]
+    )
+
+    return await delayedInbox.functions.sendL2Message(sendData)
+  }
+
+  /**
+   * Sign a transaction with msg.to, msg.value and msg.data.
+   * You can use this as a helper to call inboxTools.sendL2SignedMessage
+   * above.
+   * @param message A signed transaction which can be sent directly to network,
+   * tx.to, tx.data, tx.value must be provided when not contract creation, if
+   * contractCreation is true, no need provide tx.to. tx.gasPrice and tx.nonce
+   * can be overrided.
+   * @param contractCreation this transaction is used to create contract or not.
+   * @param l2Signer ethers Signer type, used to sign l2 transaction
+   * @returns The l1 delayed inbox's transaction signed data.
+   */
+  public async signL2Tx(
+    tx: TransactionRequest,
+    contractCreation: boolean,
+    l2Signer: Signer
+  ): Promise<string> {
+    //check required args
+    if (!((tx.to || contractCreation) && tx.data && tx.value)) {
+      throw new ArbSdkError('Required arg not provided')
+    }
+
+    if (!tx.nonce) {
+      tx.nonce = await l2Signer.getTransactionCount()
```

Comment:
It's better not to mutate objects that are passed in to a function. This can result in unexpected things happening for the caller - they may want to use the transaction request elsewhere and be surprised when the properties have changed. Better to create a new object and populate it from the argument